dags/trigger_k8s_cronjob.py

```
@@ -0,0 +1,90 @@
+    except ApiException as e:
+      print("Exception when calling BatchV1Api->create_namespaced_job: %s\n" % e)
+
+    # Get the uid from the newly created job
+    controllerUid = created_job.metadata.uid
+
+    core_v1 = client.CoreV1Api(client.ApiClient(configuration))
+
+    # Create a label_selector from the job's UID
+    pod_label_selector = "controller-uid=" + controllerUid
+    try:
+      # Get the pod name for the newly created job
+      pods_list = core_v1.list_namespaced_pod(namespace, label_selector=pod_label_selector, timeout_seconds=10)
+      pod_name = pods_list.items[0].metadata.name
+    except ApiException as e:
+      print("Exception when calling CoreV1Api->list_namespaced_pod: %s\n" % e)
+
+
+    try:
+      # Get the status of the newly created job
+      status = core_v1.read_namespaced_pod_status(pod_name, namespace).status.phase
+    except ApiException as e:
+      print("Exception when calling CoreV1Api->read_namespaced_pod_status: %s\n" % e)
+
+    # Sleep while the pod has not completed, break on Failed or Succeeded status
+    while status == 'Pending' or status =='Running':
+      try:
+        status = core_v1.read_namespaced_pod_status(pod_name, namespace).status.phase
+        print('Current Status: ' + status)
+        if status == 'Succeeded' or status == 'Failed':
+          break
+        print('sleeping')
+        time.sleep(5)
+      except ApiException as e:
+        print("Exception when calling CoreV1Api->read_namespaced_pod_status: %s\n" % e)
+
+    try:
+      # Retrieve and print the log from the finished pod
+      pod_log = core_v1.read_namespaced_pod_log(name=pod_name, namespace=namespace, follow=True, pretty=True, timestamps=True)
+      print(pod_log)
+    except ApiException as e:
+      print("Exception when calling CoreV1Api->read_namespaced_pod_log: %s\n" % e)
+    print(status)
+    # Return True if status='Succeeded', False if 'Failed'
+    if status == 'Succeeded':
+      return True
+    return False
```

Looks like the return value gets printed in the logs when using the PythonOperator, we might want to throw an exception/ return 'Job succeeded' instead

https://airflow.apache.org/docs/stable/howto/operator/python.html